packages/playback-core/src/index.ts

```diff
@@ -230,12 +228,66 @@ export const generatePlayerInitTime = () => {
 
 export const generateUUID = mux.utils.generateUUID;
 
-export const toMuxVideoURL = (playbackId?: string, { domain = MUX_VIDEO_DOMAIN, maxResolution = '' } = {}) => {
-  if (!playbackId) return undefined;
-  const [idPart, queryPart = ''] = toPlaybackIdParts(playbackId);
-  const url = new URL(`https://stream.${domain}/${idPart}.m3u8${queryPart}`);
-  if (maxResolution) {
-    url.searchParams.set('max_resolution', maxResolution);
+type MuxVideoURLProps = Partial<{
+  playbackId: string;
+  customDomain: string;
+  maxResolution: MaxResolutionValue;
+  minResolution: MinResolutionValue;
+  renditionOrder: RenditionOrderValue;
+  tokens: Partial<{
+    playback: string;
+    storyboard: string;
+    thumbnail: string;
+  }>;
+  extraPlaylistParams: Record<string, any>;
+}>;
+
+export const toMuxVideoURL = ({
+  playbackId: playbackIdWithParams,
+  customDomain: domain = MUX_VIDEO_DOMAIN,
+  maxResolution,
+  minResolution,
+  renditionOrder,
+  tokens: { playback: token } = {},
+  extraPlaylistParams = {},
+}: MuxVideoURLProps = {}) => {
+  if (!playbackIdWithParams) return undefined;
+  const [playbackId, queryPart = ''] = toPlaybackIdParts(playbackIdWithParams);
+  const url = new URL(`https://stream.${domain}/${playbackId}.m3u8${queryPart}`);
+  /*
+   * All identified query params here can only be added to public
+   * playback IDs. In order to use these features with signed URLs
+   * the query param must be added to the signing token.
+   *
+   * */
+  if (token || url.searchParams.has('token')) {
+    url.searchParams.forEach((_, key) => {
+      if (key != 'token') url.searchParams.delete(key);
+    });
+    if (token) url.searchParams.set('token', token);
```

Comment:
here's some unneeded work done I think.
either drop this line `if (token) url.searchParams.set('token', token);`
OR reset the whole `url.searchParams = new URLSearchParams()` and leave the set token line

Reply:
`url.searchParams` is read only 💔 I actually tried that for the `forEach()` that deletes any additional query params above. We could do **_different_** work to replace the `url.searchParams` with a refactor for sure. One complexity though is that in our current component implementations, there are two different ways of receiving a token value:
1. the `token` value may come in via `playbackId=`MYPLAYBACKID?token=MYTOKEN`
2. the `token` value may come in via `token.playback=MYTOKEN`
3. the `token` may be specified on both (in which case we should have some rule on behavior)
We could certainly refactor the conditional paths also, but we will still need to set the `token` in some form for case at least case (2) (here, it's 2 & 3, mostly because that was my S1 on what the rule should be). Lmk your thoughts here.